DOC: Remove Editor module documentation

jamesobutler opened the pull request:
re #5664

The Editor module was removed in https://github.com/Slicer/Slicer/commit/39283db420baf502fa99865c9d5d58d0e5295a6e. The Segment Editor module was the replacement.

lassoan reviewed (dismissed):
Thank you, these changes look good to me.

jcfr commented:
Similarly to what we do for extension in  https://github.com/Slicer/ExtensionsIndex/tree/master/ARCHIVE, I  think we should include a section in the documentation to list the past modules.

What do you think of adding a section `Modules / Retired Modules` or  `Modules / Retired`

| `Modules / Retired Modules` |  `Modules / Retired` |
|--|--|
| ![image](https://user-images.githubusercontent.com/219043/143158629-24dcd847-46f9-467c-96d4-9f76d297d05e.png) |   ![image](https://user-images.githubusercontent.com/219043/143158705-ff73804d-377f-4cb0-b747-8558cdc95543.png) |

jcfr commented on Docs/user_guide/modules/retiredmodules.md:
```diff
@@ -0,0 +1,12 @@
+A short list of reasons why a module may have been retired
+
+- The module has been superseded with another, and is not recommended
+- The module has no development support, and no longer runs properly
```
The list does not seem to be rendered properly:

![image](https://user-images.githubusercontent.com/219043/143253404-0681a066-30be-4445-aa0a-acabf7491e6f.png)

jamesobutler replied:
I noticed that as well. Not sure why. I'll push again to see if something changes.

jamesobutler replied:
With latest push they are still not rendering appropriately. Not sure why

jcfr reviewed (commented):
Thanks, looks good to me.

jamesobutler commented:
A retired modules page has been added.
![image](https://user-images.githubusercontent.com/15837524/143254967-a9d425d3-ed5b-4dec-8a26-f64cfc22db15.png)

I've also updated the "Legacy" category in the documentation to include more modules based on what has already been set as Legacy in the Slicer application
![image](https://user-images.githubusercontent.com/15837524/143255172-796549c6-6fad-4240-bfdd-3e599a812cc3.png)

The documentation had "Label Statistics" module under the legacy category, but in the Slicer application it was not under the legacy category. I've added a commit to move it to the legacy category to match.

jcfr commented:
Thanks for working on consolidating the documentation :pray:. It will  help reduce confusion from users.

Since you are working on this, what do you think of updating the module page of each legacy module to include a warning section with a message like this:

```
This module is deprecated and will be removed in the future. To implement new workflow, we recommend to use <name-and-link-to-alternative-module>.
```

When possible and available, it may also be worth linking to the discussion page where we last discussed the upcoming "retirement" (aka move to legacy) of the module.

jcfr commented:
All of that said, moving the commits listed below to a new PR called `Add retired modules page and consolidate legacy category` may be sensible. That way we could integrate this PR that was originally intended to update Editor documentation.

```
DOC: Add retired modules page
DOC: Move legacy modules to legacy category
ENH: Move Label Statistics to Legacy category 
```

jamesobutler commented:
New PR for the legacy and retired modules documentation work: https://github.com/Slicer/Slicer/pull/6039
